### ISSUE TITLE
[Bug]: The slow generation that occurs after running the dataset-worker for a few hours

### ISSUE THREAD

**Issue by jyma** [data-preservation-programs/singularity#358]
### Description

```text
When I use V2 in China and execute the command `singularity run dataset-worker --concurrency=60` , after a certain period of time (approximately a few hours), the network bandwidth usage drops significantly.
It returns to normal after I restart the dataset-worker.
Additionally, I never encountered this issue when using V1.
```

### Steps to Reproduce

1. Run below commands
```
singularity run dataset-worker --concurrency=60
```
2. Wait a few hours
3. See the network bandwidth usage drops significantly.(Maybe  this issue only occurs in mainland China's network environment)

### Version

0.5.0

### Operating System

Linux

### Database Backend

SQLite

**Comment by xinaxu** [comment]
@jyma what is the name of the dataset you're preparing and the CLI command you used to create the storage connection.

**Comment by jyma** [comment]
singularity storage create s3 aws --name "4DN" --path "4dn-open-data-public" --client-scan-concurrency=60
singularity prep create --source "4DN" --name "4DN-prep" --max-size "18GiB"  --local-output /storage1/4DN

**Comment by xinaxu** [comment]
@jyma could you confirm if this is fixed?

**Comment by jyma** [comment]
@xinaxu  it had fixed in v0.5.3

**Comment by xinaxu** [comment]
Could you file another issue describing the that?

**Comment by gammazero** [comment]
@jyma I think the fix you are looking for is in v0.5.4.
